client/src/main/java/io/pravega/client/control/impl/Controller.java

```diff
@@ -118,6 +118,46 @@
      */
     CompletableFuture<Boolean> updateStream(final String scope, final String streamName, final StreamConfiguration streamConfig);
 
+    /**
+     * API to add a Subscriber to the Stream.
+     * @param scope Scope name
+     * @param streamName Stream name
+     * @param subscriber Name/Id that uniquely identifies a Stream Subscriber.
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the subscriber was updated in Stream Metadata.
+     */
+    CompletableFuture<Boolean> addSubscriber(final String scope, final String streamName, final String subscriber);
```

Comment:
why is it a boolean? is boolean helpful to user if the subscriber was already present?

Reply:
All APIs in this interface return boolean indicating the operation either failed or succeeded...The ControllerImpl implementation returns a true/false based on weather the operation was successful or not...
For "addSubscriber" the operation is successful if subscriber was not present and got added, in every other case its a failure....

Reply:
semantics of true false and error cases could be consistent with create stream api:
true - if its newly created
false - if it already exists
exception - specific to the error occurred.

Reply:
Changed as per createStream.